Cromwell server continually throws the same exception

I have a wdl task that doesn't handle an optional parameter as it should.
The wdl validates (wdltool-0.8.jar), but when I submit to my server, an exception is thrown over and over again.


```
[ERROR] [01/10/2017 15:07:12.214] [cromwell-system-akka.actor.default-dispatcher-5] [akka://cromwell-system/user/cromwell-service/WorkflowManagerActor/WorkflowActor-98777f84-7e
14-4408-b31e-9b57db5d813b/WorkflowExecutionActor-98777f84-7e14-4408-b31e-9b57db5d813b/98777f84-7e14-4408-b31e-9b57db5d813b-EngineJobExecutionActor-snps.testHC:NA:1/98777f84-7e1
4-4408-b31e-9b57db5d813b-BackendJobExecutionActor-98777f84:snps.testHC:-1:1/DispatchedConfigAsyncJobExecutionActor] DispatchedConfigAsyncJobExecutionActor [UUID(98777f84)snps.t
estHC:NA:1]: Error attempting to Execute
java.lang.UnsupportedOperationException: Could not find declaration for WdlOptionalValue(WdlStringType,None)
        at wdl4s.command.ParameterCommandPart.instantiate(ParameterCommandPart.scala:48)
        at wdl4s.Task$$anonfun$instantiateCommand$1$$anonfun$apply$2.apply(Task.scala:108)
        at wdl4s.Task$$anonfun$instantiateCommand$1$$anonfun$apply$2.apply(Task.scala:108)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
        at scala.collection.Iterator$class.foreach(Iterator.scala:893)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
        at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
        at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
        at scala.collection.AbstractTraversable.map(Traversable.scala:104)
        at wdl4s.Task$$anonfun$instantiateCommand$1.apply(Task.scala:108)
        at wdl4s.Task$$anonfun$instantiateCommand$1.apply(Task.scala:108)
        at scala.util.Try$.apply(Try.scala:192)
        at wdl4s.Task.instantiateCommand(Task.scala:108)
...
```


Restarting the service seems to be the only way of stopping the exception being thrown.
Would it be possible to log the exception just once so that it doesn't clog up the server logs?
Thanks

--- a/backend/src/main/scala/cromwell/backend/async/AsyncBackendJobExecutionActor.scala
+++ b/backend/src/main/scala/cromwell/backend/async/AsyncBackendJobExecutionActor.scala
@@ -1,5 +1,6 @@
 package cromwell.backend.async
 
+
 import akka.actor.{Actor, ActorLogging, ActorRef}
 import cromwell.backend.BackendJobDescriptor
 import cromwell.backend.BackendJobExecutionActor._
@@ -35,7 +36,15 @@ trait AsyncBackendJobExecutionActor { this: Actor with ActorLogging =>
 
   def retryable: Boolean
 
-  def isFatal(throwable: Throwable): Boolean = throwable.isInstanceOf[CromwellFatalExceptionMarker]
+  // The scala package object (scala/package.scala) contains a neat list of runtime errors that are always going to be fatal.
+  // We also consider any Error as fatal, and include the CromwellFatalExceptionMarker so we can mark our own fatal exceptions.
+  def isFatal(throwable: Throwable): Boolean = throwable match {
+    case _: Error => true
+    case _: RuntimeException => true
+    case _: InterruptedException => true
+    case _: CromwellFatalExceptionMarker => true
+    case _ => false
+  }
 
   def isTransient(throwable: Throwable): Boolean = !isFatal(throwable)
 

--- a/backend/src/test/scala/cromwell/backend/BackendSpec.scala
+++ b/backend/src/test/scala/cromwell/backend/BackendSpec.scala
@@ -52,7 +52,7 @@ trait BackendSpec extends ScalaFutures with Matchers with Mockito {
                                           runtimeAttributeDefinitions: Set[RuntimeAttributeDefinition]): BackendJobDescriptor = {
     val call = workflowDescriptor.workflow.taskCalls.head
     val jobKey = BackendJobDescriptorKey(call, None, 1)
-    val inputDeclarations = call.evaluateTaskInputs(inputs, NoFunctions)
+    val inputDeclarations = call.evaluateTaskInputs(inputs, NoFunctions).get // .get is ok because this is a test
     val evaluatedAttributes = RuntimeAttributeDefinition.evaluateRuntimeAttributes(call.task.runtimeAttributes, NoFunctions, inputDeclarations).get // .get is OK here because this is a test
     val runtimeAttributes = RuntimeAttributeDefinition.addDefaultsToAttributes(runtimeAttributeDefinitions, options)(evaluatedAttributes)
     BackendJobDescriptor(workflowDescriptor, jobKey, runtimeAttributes, inputDeclarations)

--- a/engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/JobPreparationActor.scala
+++ b/engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/JobPreparationActor.scala
@@ -32,20 +32,19 @@ abstract class CallPreparationActor(val workflowDescriptor: EngineWorkflowDescri
   }
 
   def resolveAndEvaluateInputs(): Try[Map[Declaration, WdlValue]] = {
-    Try {
-      val call = callKey.scope
-      val scatterMap = callKey.index flatMap { i =>
-        // Will need update for nested scatters
-        call.ancestry collectFirst { case s: Scatter => Map(s -> i) }
-      } getOrElse Map.empty[Scatter, Int]
 
-      call.evaluateTaskInputs(
-        workflowDescriptor.backendDescriptor.knownValues,
-        expressionLanguageFunctions,
-        outputStore.fetchNodeOutputEntries,
-        scatterMap
-      )
-    } recoverWith {
+    val call = callKey.scope
+    val scatterMap = callKey.index flatMap { i =>
+      // Will need update for nested scatters
+      call.ancestry collectFirst { case s: Scatter => Map(s -> i) }
+    } getOrElse Map.empty[Scatter, Int]
+
+    call.evaluateTaskInputs(
+      workflowDescriptor.backendDescriptor.knownValues,
+      expressionLanguageFunctions,
+      outputStore.fetchNodeOutputEntries,
+      scatterMap
+    ).recoverWith {
       case t: Throwable => Failure(new VariableLookupException(s"Couldn't resolve all inputs for ${callKey.scope.fullyQualifiedName} at index ${callKey.index}.", List(t)))
     }
   }

--- a/engine/src/test/scala/cromwell/OptionalParamWorkflowSpec.scala
+++ b/engine/src/test/scala/cromwell/OptionalParamWorkflowSpec.scala
@@ -27,10 +27,8 @@ class OptionalParamWorkflowSpec extends Matchers with WordSpecLike {
         fail("Expected to find task 'find'")
       }
 
-      val instantiateWithoutValue = findTask.instantiateCommand(findTask.inputsFromMap(Map("find.root" -> WdlFile("src"))), NoFunctions) getOrElse {
-        fail("Expected instantiation to work")
-      }
-      instantiateWithoutValue shouldEqual "find src"
+      val instantiateWithoutValue = findTask.instantiateCommand(findTask.inputsFromMap(Map("find.root" -> WdlFile("src"))), NoFunctions)
+      instantiateWithoutValue.get shouldEqual "find src"
 
       val instantiateWithValue = findTask.instantiateCommand(findTask.inputsFromMap(Map(
         "find.root" -> WdlFile("src"),

--- a/project/Dependencies.scala
+++ b/project/Dependencies.scala
@@ -2,7 +2,7 @@ import sbt._
 
 object Dependencies {
   lazy val lenthallV = "0.20"
-  lazy val wdl4sV = "0.9"
+  lazy val wdl4sV = "0.10-3ad63bf-SNAP"
   lazy val sprayV = "1.3.3"
   /*
   spray-json is an independent project from the "spray suite"

